### PR TITLE
Integration tests and FSTUserDataConverter fixes for array transforms.

### PR DESCRIPTION
**Pull request by mikelehen**

### CODE REVIEW

File: Firestore/Example/Tests/Util/FSTIntegrationTestCase.mm [hunk 1]
```diff
@@ -276,6 +276,13 @@ - (void)deleteDocumentRef:(FIRDocumentReference *)ref {
   [self awaitExpectations];
 }
 
+- (void)mergeDocumentRef:(FIRDocumentReference *)ref data:(NSDictionary<NSString *, id> *)data {
+  [ref setData:data
+           merge:TRUE
```
Comment (wilhuff) [inline Firestore/Example/Tests/Util/FSTIntegrationTestCase.mm]: s/TRUE/YES/

File: Firestore/Source/API/FSTUserDataConverter.mm [hunk 2]
```diff
@@ -515,6 +515,13 @@ - (nullable FSTFieldValue *)parseData:(id)input context:(FSTParseContext *)conte
   input = self.preConverter(input);
   if ([input isKindOfClass:[NSDictionary class]]) {
     return [self parseDictionary:(NSDictionary *)input context:context];
+
+  } else if ([input isKindOfClass:[FIRFieldValue class]]) {
+    // parseSentinelFieldValue may add an FSTFieldTransform, but nothing should be included in
+    // the actual parsed data, so we don't add this path to our fieldMask and we return nil.
+    [self parseSentinelFieldValue:(FIRFieldValue *)input context:context];
```
Comment (wilhuff) [inline Firestore/Source/API/FSTUserDataConverter.mm]: This code is no longer calling `[context appendToFieldMaskWithFieldPath:*context.path]` which doesn't seem right.

If I `update(["a.b": FieldValue.serverTimestamp()])` shouldn't there be a field mask for "a.b"?

Comment (wilhuff) [reply Firestore/Source/API/FSTUserDataConverter.mm]: That makes sense. Thanks for explaining.

**Comment by mikelehen** [comment]
I'll re-target to master and merge once the prereq PRs land...

**Pull request merged**